File: examples/supplychain-app/fabric/chaincode_rest_server/chaincode/common/UpdateRequest.go
package common

// The UpdateRequest models a product update in a supply chain
type UpdateRequest struct {
	ID       string                 `json:"trackingID"`
	Health   string                 `json:"health"`
	Metadata map[string]interface{} `json:"misc"`
	Location string                 `json:"lastScannedAt"`
}

File: examples/supplychain-app/fabric/chaincode_rest_server/rest-server/src/models/UpdateRequest.ts
import { deserialize } from 'class-transformer';
import { IsDefined, IsNotEmpty, IsUUID } from 'class-validator';
import { Validateable } from 'mixins/Validateable';

export class UpdateRequest extends Validateable {
  static fromJSON(json: string): UpdateRequest {
    return deserialize(UpdateRequest, json);
  }

  @IsNotEmpty()
  @IsDefined()
  health: string;

  @IsNotEmpty()
  @IsDefined()
  misc: { [key: string]: any };

  @IsNotEmpty()
  @IsDefined()
  @IsUUID()
  trackingID: string;

  @IsNotEmpty()
  @IsDefined()
  location: string;
}
